SARAALERT-888: Monitoring status refactor

sgober opened the pull request:
# Description
Jira Ticket: [SARAALERT-888](https://tracker.codev.mitre.org/browse/SARAALERT-888)

Refactor of the MonitoringStatus component.  This includes:
- Create a parent MonitoringActions component
- Break out MonitoringStatus into separate components for each piece. If there are pieces that can use a general dropdown component, do so.
- Clean up code, comment, reduce any duplication/confusion, etc.
- Add frontend tests for all of these.
- Two special cases regarding LDE section of the monitoring status modal (see JIRA)

# Important Changes
Please list important files (meaning substantial or integral to the PR) along with a list of the general changes that should be highlighted for reviewers.

**There are A LOT of changes here, so I'm going to break this down by each change and what files are effected by those changes**

1. Created a parent Monitoring Actions component: `MonitoringActions.js`, `show.html.erb`

2. Broke each Monitoring Action into child components: `AssignedUser.js`, `Jurisdiction.js`, `GenericAction.js`

3. Simplified existing child components of Monitoring Actions: `MonitoringStatus.js`, `CaseStatus.js`

4. Only display the LDE update section in the Monitoring Status component IF the monitoree has a household member that is in the exposure workflow with continuous monitoring: `MonitoringStatus.js`, `patients_controller.rb`, `show.html.erb`

5. Updated the "this monitoree and all household members” option in the Monitoring Status to include "This monitoree and household members (this will turn OFF continuous exposure for all household members)" and hide the LDE section IF this option is selected: `MonitoringActions.js`

6. Updated @group_members and @all_group_members backend variables to be @dependents_exclude_hoh and @household_members respectively to be much more clear: `patients_controller.rb`, `show.html.erb`, `edit.html.erb`

7. Modified has_group_members props variable name to has_dependents for clarity: `show.html.erb`, `edit.html.erb,` `new.html.erb`, `new_group_member.html.erb`, `MonitoringActions.js`, `AssignedUser.js`, `CaseStatus.js`, `GenericAction.js`, `Jurisdiction.js`, `MonitoringStatus.js`, `Enrollment.js`, `Exposure.js`, `PatientPage.js` (dependents instead of has_dependents)

8. Modified in_a_group props variable name to is_household_member for clarity: `show.html.erb`, `LastDateExposure.js`

9. Added @household_members_with_ce_in_exposure variable to the backend to serve the very specific case of #4 above: `MonitoringStatus.js`, `patients_controller.rb`, `show.html.erb`

10. Changed name = jurisdictionId or assignedUser to id = jurisdiction_id or assigned_user to be consistent with the rest of the monitoring actions:  `AssignedUser.js`, `Jurisdiction.js`,  `Exposure.js`, `PatientPage.js`, `PatientsTable.js` plus the tests modified in #15

11. Removed unused code for isolation and pause notifications: `MonitoringActions.js`, `MonitoringStatus.js`, `show.html.erb`

12. Fixed datepicker styling: `MonitoringStatus.js`, `ExtendedIsolation.js`

13. Added react component tests for all new components: `MonitoringActions.test.js`, `AssignedUser.test.js`, `GenericAction.test.js`, `Jurisdiction.test.js`,  `MonitoringStatus.test.js`

14. Updated react component tests/mocks for clarity and additional use cases as well as fixed tests that broke from these changes: `CaseStatus.test.js`, `PatientPage.test.js`, `MockPatients.js`

15. Fix system tests that broke in the process of this refactor: `jurisdictions_controller_test.rb`, `dashboard_verifier.rb`, `form_validator.rb`, `dashboard_verifier.rb`, `actions.rb`, `public_health_import_export_test.rb`

# Testing
This fix was tested on the following browsers (submitter must check all those that apply):
* [x] Chrome
* [ ] Firefox
* [ ] Safari
* [ ] IE11

Please describe the tests needed to verify your changes. Provide instructions for reproducing these tests. List any relevant details for your test configuration.

sgober commented:
@holmesie @kylietmo @timwongj @DPC15038 I was hoping to get as many eyes on this as possible before I start writing the tests.  Please review to make sure none of the functionality changed or is broken, as well as for additional improvements that should be made during this refactor.  This only includes the first two bullets listed above, but you'll see it's already pretty significant.

@timwongj - can you take a look at the failing tests as well?

timwongj commented on app/controllers/patients_controller.rb:
```diff
@@ -412,7 +412,7 @@ def update_fields(patient, params, household, propagation)
 
     # Update the patient with updated values.
-    patient.update(params.require(:patient).permit(params_to_update))
+    patient.update(params.permit(params_to_update))
```
I think calling `params.permit(params_to_update)` simply gets you `params[:param_1]`, `params[:param_2]`, etc. whereas `params.require(:patient).permit(params_to_update)` gets you `params[:patient][:param_1]`, `params[:patient][:param_2]`, etc.

so I think we might have to change this to `param[:patient].permit(params_to_update)` but I'm not sure if we can call `.permit` after doing `param[:patient]`, but maybe give this a try?

@holmesie thoughts?

holmesie replied:
You can actually nest `permit` calls (example: `params.permit(person: [ :name, { pets: :name } ])`), take a look at https://apidock.com/rails/ActionController/Parameters/permit for reference

timwongj replied:
awesome, thanks @holmesie 

@sgober : so I think we would be able to do something like: `params.permit(patient: params_to_update)`

sgober replied:
this works for updating the jurisdiction but none of the other monitoring actions (500 error).  not totally sure how we should proceed here

holmesie replied:
You should make this method support both types of updates, or move it to a new route.

sgober replied:
I ended up just adding patient to the update status call for jurisdictions.  seemed better than messing around with different cases on the backend

DPC15038 commented on app/javascript/components/subject/AssignedUser.js:
```diff
@@ -0,0 +1,187 @@
+    super(props);
+    this.state = {
+      showAssignedUserModal: false,
+      assigned_user: props.patient.assigned_user ? props.patient.assigned_user : '',
```
Small nitpick, but I've seen it elsewhere in the code. You can do `props.patient.assigned_user || ''` here.

sgober replied:
done

DPC15038 commented on app/javascript/components/subject/GenericAction.js:
```diff
@@ -0,0 +1,235 @@
+      this.setState({ [event.target.name]: applyToGroup });
+    } else if (event?.target?.id) {
+      let value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+      this.setState({ [event.target.id]: event?.target?.value ? value : '' });
```
Shouldn't these existence-operators (`?`) be on line 83 above? Feels like this ternary/setState bit could be re-written slightly.

sgober replied:
done

sgober commented:
This should be good for review now! @holmesie @kylietmo @timwongj @DPC15038

kylietmo commented on app/javascript/components/subject/GenericAction.js:
```diff
@@ -0,0 +1,235 @@
+    super(props);
+    this.state = {
+      patient: props.patient,
+      showExposureRiskAssessmentModal: false,
```
So I think we have two options with this component:

1. Break it out into separate components completely because it's not really generic enough to justify putting it all in one.
2. Make this component even more generic so that it can be used for almost all of the monitoring actions and doesn't contain so much action-specific code. We might still want there to be a new component for every action, but there would be much less code in each since the skeleton component would contain a good chunk of it. 

I'd much prefer #2 if we think it's do-able. Here's what I'm thinking:
- Every component in the Monitoring Actions is a dropdown with a modal. Make this a "dumb" (presentational) generic component `MonitoringStatusDropdown` that just has that skeleton (modal with checkboxes, change handler(s), dropdown). 
- Everything else should be handled by props. You can pass routes as props, you can pass the handlers as props if need be as well. We may still want wrapper components around these to pass the necessary props. 
Basically taking advantage of [this pattern](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0).
Thoughts? I just don't think this as it is is super useful - we're running into the same problem as before where it's too much in one.

sgober replied:
My thoughts on that is I feel like #2 would become too much like we had before so it might make sense to just do individualized ones...  I know neither is preferable but I do feel like with #1 it would be MUCH easier to follow and definitely cleaner.  Let me know which you're thinking and I can change it.  Going to make the other changes for now

kylietmo replied:
I think we can get the clarity of #1 with the removal of duplication of #2. 

Here's what I'm thinking, let me know how this sounds. I'll use `AssignedUser` as an example.
- New component `MonitoringActionsDropdown` that contains the following:
    - Contains you already have in `AssignedUser.js` (createModal, submit, render, toggleModal, handleKeyPress) EXCEPT you make it not specific to AssignedUser, and you accept the POST route, handlers, and state change callback(s) as props. 
- Update component `AssignedUser.js` that  maintains the current non-presentational state logic (i.e. `apply_to_group`, `assigned_user`, `reasoning`, etc.) that it's already maintaining, but instantiates the `MonitoringActionsDropdown` with the correct props, and this component manages presentation logic such as `loading` and `showModal`.

 This way `AssignedUser` is still separate and still handles all the logic, but `MonitoringActionsDropdown` handles all the repeated modal, dropdown, and submission logic. Kind of like a new util or layout component. It'd be similar to the `CustomTable` presentational component that's being used by the `PatientsTable` and the `AdminTable`

Do you foresee any issues with this that I may be overlooking?

sgober replied:
The issue that I think might be the big one is then we're getting back to the point where the createModal is going to be a nightmare of logic and html (like it was before).  I get not wanting to repeat code but I just can't see having a generic createModal that would be better than before...

Maybe I'm not fully understanding but I don't think that would be really much better than we had before, perhaps less confusing by a bit.

The other thing is that Jurisdiction and Assigned User are different than the other actions and would therefore still have to be separate on a rendering level

sgober replied:
Thinking about this some more... a good amount of the Case/Monitoring Status modal logic is handled by state change and is more complicated.  Factor that in with the fact that the jurisdiction/assigned user inputs are different then you're looking at Monitoring Plan, Public Health Action and Exposure Risk Assessment which I originally broke out into that generic copmponent.

Long answer but I'm not sure there's a good answer here.  If we keep it too general, it may start looking similar to what it was before, but we definitely are repeating code by separating them.  this would be good though if some of these become more custom than they already are (like how case status got)

kylietmo replied:
Well the idea would be that that logic handling would still be done in the `CaseStatus` component (for example), and the generic component would just handle displaying the modal and calling the correct callbacks which would be in props.`createModal` shouldn't have to deal with any logic. 

I do think this would be do-able without getting logic mixed up because the generic component would be purely presentational, but I don't think it's a necessary part of this refactor for now. And if they do get more complex for any reason this does make it more straightforward, so I'm okay with just splitting it out here - which I think you already did, so I can resolve this thread.

sgober replied:
With all the cases the modal currently has I feel like that would be difficult - however once the PO team addresses how they want to handle everything moving forward that might get easier

kylietmo commented on app/javascript/components/subject/Jurisdiction.js:
```diff
@@ -0,0 +1,202 @@
+              onKeyPress={this.handleKeyPress}
+              value={this.state.jurisdiction_path}
+            />
+            <datalist id="jurisdictionPaths">
```
Is it time to stop using `datalists` here and convert to the React dropdowns you use elsewhere?

sgober replied:
I'm going to say no on this one mainly because there is a LOT of manipulating that needs to happen to get those to work.  They're great if you need advanced functionality for the dropdown (typing, multi select, etc) but these are so basic it would likely be more trouble than it is worth to do

kylietmo replied:
That's fine with me - we can consider that out of scope of this task then. Would still be nice to do eventually I think though, since datalists aren't great looking. I can put something on the backlog.

kylietmo commented on app/javascript/components/subject/Jurisdiction.js:
```diff
@@ -0,0 +1,202 @@
+              })}
+            </datalist>
+            {!this.state.validJurisdiction || this.state.jurisdiction_path === this.props.jurisdictionPaths[this.state.original_jurisdiction_id] ? (
+              <Button className="btn-lg btn-square text-nowrap ml-2" disabled>
```
Same feedback as elsewhere about just putting this check on the `disabled` prop.

sgober replied:
done

kylietmo reviewed (changes requested):
This refactor was very much needed - so thank you! Also great addition of the tests, those will be awesome to have. 

I have a few smaller comments and then one main one (which I detailed more in comment below) that's about the fact that all of these components have the same structure (dropdown with a confirmation modal and a post to the backend) so I think they can all use a "dumb" generic component.

kylietmo commented on app/javascript/components/subject/MonitoringActions.js:
```diff
@@ -0,0 +1,109 @@
+    return (
+      <React.Fragment>
+        <Form className="mb-3 mt-3 px-4">
+          <Row>
```
Oh another comment I forgot to add, I don't think this `Row` or the `Col` are necessary here.

sgober replied:
done

DPC15038 commented on app/javascript/components/subject/AssignedUser.js:
```diff
@@ -0,0 +1,187 @@
+              })}
+            </datalist>
+            {this.state.assigned_user === this.state.original_assigned_user ? (
+              <Button className="btn-lg btn-square text-nowrap ml-2" disabled>
```
I had the same thought. I dont hate the way it currently is though. It's very readable. But this^ change probably is better/DRYer.

sgober replied:
done

sgober commented on app/controllers/patients_controller.rb:
```diff
@@ -28,12 +28,13 @@ def show
     dependents = current_user.get_patient(@patient.responder_id)&.dependents
-    @all_group_members = ([@patient] + (dependents.nil? ? [] : dependents)).uniq
+    @household_members = ([@patient] + (dependents.nil? ? [] : dependents)).uniq
+    @household_members_with_ce_in_exposure = dependents.nil? ? [] : dependents.where(isolation: false, continuous_exposure: true)
```
this is used for the LDE update in the monitoring status modal which does not include the current monitoree, so that's not needed. if it's unclear we can rename it

kylietmo replied:
Ah okay, yeah maybe call it `dependents_with_ce_in_exposure`.

sgober replied:
The thing with this is that it's not HoH dependent, which is what I've been using "dependents" for rather than household members.  However I can see the proceeding line is now confusing in that regard.  I will swap to that variable name

kylietmo replied:
What do you mean "not HoH dependent"?

sgober replied:
`@household_members` gets all the members of the household including the current monitoree, regardless of if they are the HoH or not whereas `@dependents_exclude_hoh` ONLY works on the HoH and will return an empty array if you run it on a dependent

kylietmo replied:
Right, so I guess I'm confused why you're saying that that is inconsistent with changing this variable to `dependents_with_ce_in_exposure` since this will be empty if the person is not a HoH.

kylietmo replied:
Okay I see the changes you made here: https://github.com/SaraAlert/SaraAlert/pull/450/commits/720a0e06072fbd86e67daa1f1087e4aefdba7217

Looks like there's still two instances of the `dependents` variable there that need to be updated to be `household`. I do think it's really unclear at first why `patient` is not being added to the `household_members_with_ce_in_exposure`, so I'd at least add a comment explaining that above that line.

sgober replied:
`@household_members_with_ce_in_exposure` is _**only**_ used for that update LDE section as a part of the monitoring status modal, which as you know has a very specific use case.  I can add an `excludes_patient` to the end or something but then the variable name is getting way too long. Although it is only used in that one place so I will added for clarity.  Also will add a comment

sgober replied:
all set

DPC15038 commented on app/javascript/components/subject/Jurisdiction.js:
```diff
@@ -0,0 +1,202 @@
+      loading: false,
+      reasoning: '',
+    };
+    this.origState = Object.assign({}, this.state);
```
Maybe it's just cause Im more familiar with Lodash, but I always prefer using `_.cloneDeep( value )` instead of `Object.assign` to prevent any possible data-losses with cloning. Not a huge thing (especially in this file), but it's an option.

sgober replied:
We use `Object.assign` like this in other files too so I'm going to leave this as is for now

sgober commented on app/javascript/components/enrollment/steps/Exposure.js:
```diff
@@ -585,7 +585,7 @@ class Exposure extends React.Component {
                         isInvalid={this.state.errors['jurisdiction_id']}
                         as="input"
-                        name="jurisdictionId"
+                        nid="jurisdiction_id"
```
lol whoops

DPC15038 commented on app/javascript/components/subject/Jurisdiction.js:
```diff
@@ -0,0 +1,202 @@
+  };
+
+  toggleJurisdictionModal = () => {
+    let current = this.state.showJurisdictionModal;
```
If it isnt changing, then `const` maybe? Another small thing.

sgober commented on test/system/roles/public_health/public_health_import_export_test.rb:
```diff
@@ -11,168 +11,6 @@ class PublicHealthTest < ApplicationSystemTestCase
-
-  ## Dashboard
-
-  test 'verify patient information on dashboard' do
```
this was a change made by Tim - @timwongj is this the case?

timwongj replied:
yeah I removed them because they were duplicate and I added that to the commit message but I believe that commit was squashed

sgober replied:
yeah it was - sorry @timwongj

sgober commented on app/javascript/components/subject/AssignedUser.js:
```diff
@@ -0,0 +1,187 @@
+      showAssignedUserModal: false,
+      assigned_user: props.patient.assigned_user || '',
+      original_assigned_user: props.patient.assigned_user || '',
+      apply_to_group: false,
```
done

kylietmo replied:
Question: I noticed when seeing the commit for this change that some of the places we had `apply_to_group`/`apply_to_household` apply only to dependents. Does it generally mean only apply to dependents or does it also sometimes mean apply to the entire household?

sgober replied:
apply to household always mens it's the entire household, see here: https://github.com/SaraAlert/SaraAlert/blob/monitoring-status-refactor/app/controllers/patients_controller.rb#L343

sgober replied:
the only time that it does NOT include the current patient (HoH or dependent) is here: https://github.com/SaraAlert/SaraAlert/blob/monitoring-status-refactor/app/controllers/patients_controller.rb#L330 which we could rename but I think it's descriptive enough

sgober commented on app/javascript/components/subject/AssignedUser.js:
```diff
@@ -0,0 +1,187 @@
+  };
+
+  handleKeyPress = event => {
+    if (event.which === 13 && this.state.assigned_user !== this.state.original_assigned_user) {
```
done

sgober commented on app/javascript/components/subject/AssignedUser.js:
```diff
@@ -0,0 +1,187 @@
+
+  handleChange = event => {
+    if (event?.target?.id && event.target.id === 'assigned_user') {
+      if (
```
removed with refactor of handleChange methods

sgober commented on app/views/patients/show.html.erb:
```diff
@@ -8,20 +8,19 @@
+                        has_dependents: @dependents_exclude_hoh.exists?,
+                        in_household_with_member_with_ce_in_exposure: @household_members_with_ce_in_exposure.exists?,
                         patient: @patient,
                         jurisdictionPaths: @possible_jurisdiction_paths,
```
done

sgober commented on app/javascript/components/subject/AssignedUser.js:
```diff
@@ -0,0 +1,187 @@
+    this.origState = Object.assign({}, this.state);
+  }
+
+  handleChange = event => {
```
this is now done in each individualized file

kylietmo commented:
@sgober I noticed you removed the `GenericAction` test file, I think those tests were still valuable - can you salvage it and break it up for each new component that you broke the `GenericAction` component into?

sgober commented:
> @sgober I noticed you removed the `GenericAction` test file, I think those tests were still valuable - can you salvage it and break it up for each new component that you broke the `GenericAction` component into?

@kylietmo I saved it locally, just haven't had time to split it up yet.  will do that tomorrow

sgober commented:
@kylietmo @DPC15038 I **_think_** I addressed all your comments now.  Also [SARAALERT-948](https://tracker.codev.mitre.org/browse/SARAALERT-948) is now included in this PR

kylietmo reviewed (approved):
Tested and looking good to me. Also tested change for SARAALERT-948 as well. 
Would like to @timwongj to also pass through these changes before merging. There's enough changes I think it warrants some more review.

timwongj commented on app/javascript/components/public_health/PatientsTable.js:
```diff
@@ -217,7 +217,7 @@ class PatientsTable extends React.Component {
         this.updateAssignedUsers(jurisdictionId, this.state.query.scope, this.props.workflow, this.state.query.tab);
       }
-    } else if (event.target.name === 'assignedUser') {
+    } else if (event.target.name === 'assigned_user') {
```
commenting on this line because github would only allow me to comment on changed lines, but would you mind also changing `jurisdictionId` to `jurisdiction_id` as well?

sgober replied:
done

timwongj commented on app/javascript/components/public_health/PatientsTable.js:
```diff
@@ -217,7 +217,7 @@ class PatientsTable extends React.Component {
         this.updateAssignedUsers(jurisdictionId, this.state.query.scope, this.props.workflow, this.state.query.tab);
       }
-    } else if (event.target.name === 'assignedUser') {
+    } else if (event.target.name === 'assigned_user') {
```
shouldn't this be `event.target.id` now because of this change? https://github.com/SaraAlert/SaraAlert/pull/450/files#diff-605db0c430cb0318fcd9986ece737d10d16c2ab6807a7c35737484322530bb49R503

timwongj replied:
maybe something we can do as part of this refactor is change all the other inputs in the app as well from using `event.target.name` to `event.target.id`? @kylietmo @holmesie thoughts?

sgober replied:
you'd be correct - nice catch!

sgober replied:
I changed them all in this file for consistency - there's not many more that would need to be swapped

sgober replied:
The ones in `Exposure.js` and `MonitoringStatus.js` would require more of an overhaul so I will leave those for now

timwongj replied:
aren't there just 2 (assigned user and jurisdiction id) in `Exposure.js`? I see that the logic in` MonitoringStatus.js` depends on both the `name` and `id` so I agree that it would require a bit more work

sgober replied:
in `Exposure.js` id="assigned_user" and id="jurisdiction_id" are used elsewhere so probably not best to tinker with that

timwongj commented on app/views/patients/show.html.erb:
```diff
@@ -8,20 +8,19 @@
 <% end %>
 
-<%= react_component('patient/PatientPage', { current_user: current_user, patient: @patient, group_members: @group_members, patient_id: @patient.user_defined_id_statelocal || '', dashboardUrl: '/public_health', authenticity_token: form_authenticity_token, hideBody: current_user.can_modify_subject_status?, jurisdictionPath: @patient.jurisdiction[:path] }) %>
+<%= react_component('patient/PatientPage', { current_user: current_user, patient: @patient, dependents: @dependents_exclude_hoh, patient_id: @patient.user_defined_id_statelocal || '', dashboardUrl: '/public_health', authenticity_token: form_authenticity_token, hideBody: current_user.can_modify_subject_status?, jurisdictionPath: @patient.jurisdiction[:path] }) %>
```
ik it was already like this before, but could you separate this onto multiple lines (similar to monitoring actions below) so that the props are more readable?

sgober replied:
done

timwongj commented on test/controllers/jurisdictions_controller_test.rb:
```diff
@@ -99,56 +99,56 @@ def teardown; end
 
           patients = scope == 'all' ? jur.all_patients.where.not(assigned_user: nil) : jur.immediate_patients.where.not(assigned_user: nil)
-          assert_equal patients.distinct.pluck(:assigned_user).sort, JSON.parse(response.body)['assignedUsers']
+          assert_equal patients.distinct.pluck(:assigned_user).sort, JSON.parse(response.body)['assigned_user']
```
not sure if this change was accidental, but it's the reason for this test failure: https://github.com/SaraAlert/SaraAlert/pull/450/checks?check_run_id=1321034377#step:14:35

however, I think it would be nice to standardize all variable names with underscores instead of having some of them use underscores and some of them being camelCased. so I would change these to `assigned_users` (this is supposed to be plural because it's a list of existing/possible assigned users). the controller would have be changed here: https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/jurisdictions_controller.rb#L63

and if we're gonna do it for assigned users, we should do it for jurisdiction paths too: https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/jurisdictions_controller.rb#L10

the front end stuff will also need to be updated (make sure to replace all occurrences of `jurisdictionPaths` to `jurisdiction_paths` and `assignedUsers` to `assigned_users` as well as their singular form.

I think standardizing all these variables would make it much easier for us to spot any incorrect variable names and it would make sense to include these changes in a refactor like this

timwongj commented on app/javascript/components/enrollment/steps/Exposure.js:
```diff
@@ -623,21 +623,21 @@ class Exposure extends React.Component {
                         as="input"
-                        name="assignedUser"
-                        list="assignedUsers"
+                        id="assigned_user"
```
@holmesie @kylietmo I've been talking to Sarah about this, but was wondering what you guys think about using name or id to refer to inputs (just want something consistent we can go with for the whole app)

![Screen Shot 2020-10-28 at 2 13 22 PM](https://user-images.githubusercontent.com/9956561/97480405-bb337f80-1929-11eb-9d3a-c63997923735.png)

kylietmo replied:
Sorry I haven't dived into this - so is the issue that setting `id` rather than `name` results in this warning?

timwongj replied:
no worries, and to answer your question, yes. it seems like people on google are saying that this warning comes from using the same id for multiple inputs.

@sgober and I think this is strange though because although we use jurisdiction_path and assigned_user as id's in enrollment and on the patient page, they're on separate pages (not sure if these warnings are specifically referring to those fields but those are examples of fields where we are using the same id strings)

sgober replied:
I think I fixed this here: https://github.com/SaraAlert/SaraAlert/pull/450/commits/45e4d7764a1ddb395cd4d38108a1819ca1841c23

timwongj commented on app/views/patients/_patients.html.erb:
```diff
@@ -128,7 +128,7 @@
       // get a sorted, unique list of assigned users based on the scope
-      const assignedUsers = table.data()
+      const assigned_users = table.data()
                                  .filter(function (row) {
```
small thing, but can we align the function calls below?

sgober replied:
done

kylietmo commented on app/javascript/components/history/HistoryComponent.js:
```diff
@@ -21,16 +21,24 @@ class HistoryComponent extends React.Component {
       filteredHistories: this.props.histories,
       pageOfHistories: [],
     };
+    this.creatorFilterData = [
```
These changes were already merged into 1.16.0, not sure why the diff is showing them here.

sgober replied:
I'm not seeing it in 1.16.0: https://github.com/SaraAlert/SaraAlert/blob/1.16.0/app/javascript/components/history/HistoryComponent.js#L24

sgober replied:
these merge commits that are on my branch are but are not in 1.16.0: https://github.com/SaraAlert/SaraAlert/pull/450/commits

kylietmo replied:
David's PR with these changes was merged into 1.16.0 so this requires further investigation then. I'll look into what's going on with 1.16.0.
https://github.com/SaraAlert/SaraAlert/pull/471/files

kylietmo commented on app/views/patients/show.html.erb:
```diff
@@ -8,21 +8,29 @@
-                        has_group_members: @group_members.exists?,
-                        in_a_group: @all_group_members.count > 1,
+                        has_dependents: @dependents_exclude_hoh.exists?,
+                        in_household_with_member_with_ce_in_exposure: @household_members_with_ce_in_exposure_excludes_patient.exists?,
```
I'm seeing the following error when going to view a monitoree's page and I believe it's due to this line (it just highlights the wrong thing in the error message). You can't call `exists?` on an array.
![Screen Shot 2020-10-29 at 12 11 29 PM](https://user-images.githubusercontent.com/11698457/97601442-8d5d4200-19e0-11eb-9c7f-74eccd295b90.png)

sgober replied:
I'm not seeing this error - can you test on the new PR and let me know if it's still happening?
